pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/io/params/RecommenderConstants.java

```diff
@@ -0,0 +1,116 @@
+    public static final long DEFAULT_THRESHOLD_MIN_QPS_PARTITION = 200;
+    public static final long DEFAULT_OPTIMAL_SIZE_PER_SEGMENT = 2000_000_000; //2GB
+    public static final long DEFAULT_KAFKA_NUM_MESSAGES_PER_SEC_PER_PARTITION = 250;
+    public static final double DEFAULT_THRESHOLD_RATIO_MIN_DIMENSION_PARTITION_TOP_CANDIDATES = 0.8d;
+    public static final int DEFAULT_THRESHOLD_MAX_IN_LENGTH = 4;
+  }
+
+  public static class BloomFilterRule {
+    public static final long DEFAULT_THRESHOLD_MAX_CARDINALITY_BLOOMFILTER = 1000_000;
+    public static final double DEFAULT_THRESHOLD_MIN_PERCENT_EQ_BLOOMFILTER = 0.5d;
+  }
+
+  public static class NoDictionaryOnHeapDictionaryJointRule {
+    public static final double DEFAULT_THRESHOLD_MIN_FILTER_FREQ_DICTIONARY = 0d;
+    public static final double DEFAULT_THRESHOLD_MAX_SELECTION_FREQ_DICTIONARY = 0.3d;
+    public static final long DEFAULT_THRESHOLD_MIN_QPS_ON_HEAP = 10_000;
+    public static final long DEFAULT_THRESHOLD_MAX_DICTIONARY_SIZE_ON_HEAP = 1000_000L;
+    public static final double DEFAULT_THRESHOLD_MIN_FILTER_FREQ_ON_HEAP = 0.3d;
+    public static final double DEFAULT_THRESHOLD_MIN_PERCENT_DICTIONARY_STORAGE_SVAE = 0.95;
+
+    public static final double DEFAULT_DICT_COEFF_A = 0.217769;
+    public static final double DEFAULT_DICT_COEFF_B = 89.0975;
+    public static final double DEFAULT_DICT_LOWER = 0;
+    public static final double DEFAULT_DICT_UPPER = 0;
+    public static final int DEFAUlT_NUM_PARTITIONS = 16;
+
+    public static final int DEFAULT_SEGMENT_FLUSH_TIME = 86400;
+  }
+
+  public static class FlagQueryRuleParams{
+    public static final long DEFAULT_THRESHOLD_MAX_LIMIT_SIZE = 100000;
+    public static final String WARNING_NO_FILTERING = "Warning: No filtering in ths query";
+    public static final String WARNING_NO_TIME_COL = "Warning: No time column used in ths query";
+    public static final String WARNING_TOO_LONG_LIMIT = "Warning: The size of LIMIT is longer than " + DEFAULT_THRESHOLD_MAX_LIMIT_SIZE;
+    public static final String ERROR_INVALID_QUERY = "Error: query not able to parse, skipped";
+  }
+
+  public static final String PQL = "pql";
+  public static final String SQL = "sql";
+  public static final String OFFLINE = "offline";
+  public static final String REALTIME = "realtime";
+  public static final String HYBRID = "hybrid";
+  public static final int NO_SUCH_COL = -1;
+  public static final double DEFAULT_CARDINALITY = 1;
+  public static final double MIN_CARDINALITY = 1;
+  public static final double DEFAULT_AVERAGE_NUM_VALUES_PER_ENTRY = 1d;
+  public static final int DEFAULT_QPS = 100;
+  public static final int DEFAULT_LATENCY_SLA = 1000;
+  public static final int DEFAULT_NUM_MSG_PER_SEC = 250;
+  public static final int DEFAULT_NUM_RECORDS_PER_PUSH = 10000;
+  public static final int DEFAULT_INT_SIZE = Integer.BYTES;
+  public static final int DEFAULT_NULL_SIZE = 0;
```

Comment:
I don't think you need to have DEFAULT_FLOAT/INT/LONG etc size. These are fixed width columns and will always be same as specified in their respective classes. -- Float.BYTES, Long.BYTES etc

Reply:
Done